doc/user-guide/file_clients.rst

```diff
@@ -0,0 +1,87 @@
+File clients
+============
+
+.. note::
+
+    This is a guide on ``file clients``. For API docs
+    see :ref:`api-python-client`.
+
+File clients are used for uploading File products to the cloud. Currently two clients are supported for Amazon S3 and Google Cloud respectively.
+
+The upload process happens in two steps:
+
+* Given a local path, the remote path for storing the file is computed. An absolute local file path of ``/path/to/project/out/data.csv`` gets translated to ``path/to/parent/out/data.csv``. Here, ``parent`` is the parent folder in the bucket to store the files.
+* Upload the file using appropriate cloud APIs.
+
+
+Pre-requisites
+--------------
+
+* Create a bucket in the required cloud platform, or use an existing one.
+* Configure the environment with the credentials or create a `credentials.json` file if environment is not configured.
+
+
+Create a clients file
+---------------------
+
+Next, create a `clients.py` file that contains the below function for S3 client:
+
+.. code-block:: python
+
+    from ploomber.clients import S3Client
+
+    def get_s3():
+        return S3Client(bucket_name='bucket-name',
+                        parent='parent-folder-name',
+                        # pass the json_credentials_path if env not configured with credentials
+                        json_credentials_path='credentials.json')
+
+Sample file for Google Cloud Storage client:
+
+.. code-block:: python
+
+    from ploomber.clients import GCloudStorageClient
+
+    def get_gcloud():
+        return GCloudStorageClient(bucket_name='bucket-name',
+                                   parent='parent-folder-name'
+                                   # pass the json_credentials_path if env not configured with credentials
+                                   json_credentials_path='credentials.json')
+
+Configure the pipeline
+----------------------
+Now, configure the `pipeline.yaml` file to add the `clients` key to specify the S3 or GCloud function:
+
+.. code-block:: yaml
+
+   # some content
+   ......
+
+   # add this
+   clients:
+    File: project-name.clients.get_client
+
+    # content continues...
+
+
+Working with external datasets
+------------------------------
+
+The file clients only upload products generated by the pipeline. If you want to work with an external dataset which is in the cloud please follow the below approach:
+
+* Sql script that runs a CREATE TABLE statement to copy a public dataset.
```

Comment:
I'm a but confused about this section. I think this is documenting what to do when the dataset we want to work with is in a different place right? But I'm not following how that connects to this list

Reply:
This was mentioned in the original issue: `Another important concept to explain is how to bring external datasets that live in the cloud already since File clients only upload products generated by the pipeline.`

Reply:
Ok, so let's remove this three points since they're confusing and replace them with:


> The file clients only upload products generated by the pipeline. If you want to work with an external dataset, you should download such a dataset in the pipeline task that uses it as input. If you need help contact us (add link to our slack)